packages/client/src/app/components/challenge/progress/ChallengeCheckBox.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+  <div class="d-flex justify-content-center">
+    <div class="check position-relative">
+      <input
+        :id="id"
+        type="checkbox"
+        :checked="checked"
+        @change="$emit('update:checked', $event.target.checked)"
+        :disabled="isOwner ? dayNumber > todayNumber || dayNumber < todayNumber - 1 : true"
```

Comment:
it is ideal to give the template as little logic as possible. I think it's for the best to move this to a computed function? 

You can also do this for other things if you want

Reply:
Moved logic to functions